security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -567,11 +577,68 @@ func (sc *SecretCache) generateGatewaySecret(token string, connKey ConnKey, t ti
 	}, nil
 }
 
+// If there is an existing secret under a well known path, return true.
+// Otherwise, return false.
+func (sc *SecretCache) secretExists() bool {
+	b, err := ioutil.ReadFile(existingCertChainFile)
+	if err != nil || len(b) == 0 {
+		return false
+	}
+	b, err = ioutil.ReadFile(existingRootCertFile)
+	if err != nil || len(b) == 0 {
+		return false
+	}
+	b, err = ioutil.ReadFile(existingKeyFile)
+	if err != nil || len(b) == 0 {
+		return false
+	}
+
+	return true
+}
+
+// Generate a secret item from the existing secret
+func (sc *SecretCache) generateSecretFromExistingSecret(token string, connKey ConnKey) (*model.SecretItem, error) {
+	certChain, err := ioutil.ReadFile(existingCertChainFile)
+	if err != nil {
+		return nil, err
+	}
+	keyPEM, err := ioutil.ReadFile(existingKeyFile)
+	if err != nil {
+		return nil, err
+	}
+	rootCertPEM, err := ioutil.ReadFile(existingRootCertFile)
+	if err != nil {
+		return nil, err
+	}
+
+	t := time.Now()
+	var certExpireTime time.Time
+	if certExpireTime, err = nodeagentutil.ParseCertAndGetExpiryTimestamp(certChain); err != nil {
```

Comment:
Same here ( break root from key/cert ), and for root please make sure we can parse multiple certs. 

We still have the problem of cert rotation / migration / multiple roots - with SDS in pilot-agent we no longer
 have access to the config map with additional roots.

Reply:
Done.